homekit/controller/ble_impl/__init__.py

```diff
@@ -471,7 +471,7 @@ def __init__(self, pairing_data, adapter):
             # TODO Have exception here
             sys.exit(-1)
 
-        write_fun = create_ble_pair_verify_write(pair_verify_char, pair_verify_char_info['iid'])
+        write_fun = create_ble_pair_setup_write(pair_verify_char, pair_verify_char_info['iid'])
```

Comment:
so we do not have separate functions for `create_ble_pair_verify_write` and `create_ble_pair_setup_write` anymore, can we find a better name? perhaps `create_characteristic_write_function`?

Reply:
I think we can actually factor out the need for the create_ part entirely really, but i didn't want to over complicate this PR.

Now that we aren't passing it into the state machine we don't need to capture the scope. So it could just be a generic write helper like (pseudo-code, won't work as is):

```python
def write_hap_param_value(char, iid, request, expected):
    pass

...

def setup_session():
    char, iid = find_char_by_type(PAIR_VERIFY)
    state_machine = ...
    while True:
        response = write_hap_param_value(char, iid, request, expected)
        request, expected = state_machine.send(response)
```